src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    subtract(value) {
+      this.result -= value;
+
+      return this.result;
+    },
+
+    add(value) {
+      this.result += value;
+
+      return this.result;
+    },
+
+    multiply(value) {
+      this.result *= value;
+
+      return this.result;
+    },
+
+    divide(value) {
+      this.result /= value;
+
+      return this.result;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, value) {
+      this.result = callback.call(this, value);
```

Comment:
Ensure that the callback is a function before attempting to call it. You can check this using `typeof callback === 'function'` to prevent potential errors.